src/components/sectionList/filters/useSectionListFilter.ts

```diff
@@ -0,0 +1,162 @@
+import { useCallback, useMemo, useEffect, useRef } from 'react'
+import { useQueryParam, ObjectParam } from 'use-query-params'
+import {
+    Schema,
+    useSchemaFromHandle,
+    CustomObjectParam,
+    GistParams,
+} from '../../../lib'
+import { QueryRefetchFunction } from '../../../types'
+import { usePaginiationQueryParams } from '../SectionListPagination'
+
+type ObjectParamType = typeof ObjectParam.default
+
+type Filters = Record<string, string | undefined>
+
+// special key for handling search for identifiable objects
+// eg. searches for name, code, id and shortname
+// this would translate to "token" in the old API, but does not exist in GIST-API
+export const IDENTIFIABLE_KEY = 'identifiable'
+
+const IDENTIFIABLE_FIELDS = {
+    name: {
+        operator: 'ilike',
+    },
+    code: {
+        operator: 'ilike',
+    },
+    shortName: {
+        operator: 'ilike',
+    },
+    id: {
+        operator: 'eq',
+    },
+}
+
+const getVerifiedFiltersForSchema = (
+    filters: ObjectParamType,
+    schema: Schema
+): Filters => {
+    if (!filters) {
+        return {}
+    }
+    /* TODO: verify values for filters */
+    const relevantFilters = Object.entries(filters).filter(([key]) => {
+        return key === IDENTIFIABLE_KEY || schema.properties[key]
+    })
+    return Object.fromEntries(relevantFilters)
+}
+
+const useFilterQueryParam = () => {
+    return useQueryParam('filter', CustomObjectParam)
+}
+
+export const useSectionListFilters = (): ReturnType<
+    typeof useQueryParam<Filters | null | undefined, Filters>
+> => {
+    const [filter, setFilter] = useFilterQueryParam()
+    const schema = useSchemaFromHandle()
+
+    return useMemo(
+        () => [getVerifiedFiltersForSchema(filter, schema), setFilter],
+        [filter, schema, setFilter]
+    )
+}
+
+export const useSectionListFilter = (
```

Comment:
See how it's used in `ConstantSelectionFilter`. It's just a more ergonomic way to only part of the "filters", rather than the whole filter-object. So it's used in those fields since it they only operate on "eg" `domainType`. Basically to select and operate on the part you want.

`useSectionListFilters` operates on the whole `filter`-object, and is used to reset it. All other places you operate only on a certain filter.

Its basically just so you don't have do do this in every component that sets a certain filter:
```js
     if (!value) {
                setFilters((filters) => {
                    if (!filters) {
                        return undefined
                    }
                    // eslint-disable-next-line @typescript-eslint/no-unused-vars
                    const { [filterKey]: _, ...rest } = filters
                    return Object.keys(rest).length === 0 ? undefined : rest
                })
            } else {
                setFilters((filters) => ({ ...filters, [filterKey]: value }))
            }
```